### ISSUE TITLE
Consider version numbers for all/more packages in requirements.txt

### ISSUE THREAD

**Issue by awhoward** [Keck-DataReductionPipelines/KPF-Pipeline#666]
Many of the packages don't have version numbers in requirements.txt.  Should we worry about this?  For example, what happens if barycorrpy changes?

Discuss.

**Comment by awhoward** [comment]
Some packages that we might prioritize versioning: scipy, barycorrpy, KeckDRPFramework, matplotlib

Let's discuss this in the next DRP meeting.

**Comment by awhoward** [comment]
A related problem is that I tried to add a requirement that we use matplotlib version 3.8, but pip3 claimed that it doesn't know about this version, despite it being the stable release version.

**Comment by awhoward** [comment]
@bjfultn and I discussed this.  The root issue is that we adopted an earlier version of Python so many of the packages can't be upgraded without a change there.  For this reason, we're going to deprioritize this issue.  If there are cases where new features of scipy or matplotlib are really attractive, we might revisit this.

BJ also advocated for not adding version numbers to every package.  For example, we want to get the latest updates from barycorrpy when they fix a URL or make some other small change.  Many other changes from version changes with packages will be caught by the Continuous Integration.
